--- host-pkg-config-0.25.org/glib-1.2.10/glib.h	2017-01-17 14:12:23.000000000 
+++ host-pkg-config-0.25/glib-1.2.10/glib.h	2017-01-17 14:15:55.000000000 
@@ -201,13 +201,13 @@
 #    endif
 #  endif
 #endif
 #ifndef G_INLINE_FUNC
 #  ifdef __GNUC__
 #    ifdef __OPTIMIZE__
-#      define G_INLINE_FUNC extern inline
+#      define G_INLINE_FUNC static inline
 #    else
 #      undef G_CAN_INLINE
 #      define G_INLINE_FUNC extern
 #    endif
 #  else /* !__GNUC__ */
 #    ifdef G_CAN_INLINE

