--- a/generic/Config.in	2017-01-11 10:22:38.000000000 
+++ b/generic/Config.in	2017-01-21 18:50:02.000000000 
@@ -9,12 +9,38 @@
 config BR2_TARGET_GENERIC_ISSUE
        string "System banner"
        default "Welcome to Buildroot"
        help
          Select system banner (/etc/issue) to be displayed at login.
 
+choice
+	prompt "/dev management"
+	default BR2_ROOTFS_DEVICE_CREATION_STATIC
+
+config BR2_ROOTFS_DEVICE_CREATION_STATIC
+	bool "Static using device table"
+
+config BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_DEVTMPFS
+	bool "Dynamic using devtmpfs only"
+
+config BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_MDEV
+	bool "Dynamic using mdev"
+	select BR2_PACKAGE_BUSYBOX
+
+config BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_UDEV
+	bool "Dynamic using udev"
+	depends on BR2_LARGEFILE # udev
+	depends on BR2_USE_WCHAR # udev
+	select BR2_PACKAGE_UDEV
+
+comment "udev requires a toolchain with LARGEFILE + WCHAR support"
+	depends on !(BR2_LARGEFILE && BR2_USE_WCHAR)
+
+endchoice
+
+
 config CONFIG_CHIP_ID
        int "Platform Chip ID"
        default 1123
        help
          Platform ID.
 	 sun3i: 1120

